otaclient/app/ecu_info.py

```diff
@@ -11,46 +11,113 @@
-
-    def get_secondary_ecus(self):
-        return self._ecu_info.get("secondaries", [])
+DEFAULT_ECU_INFO = {
+    "format_version": 1,  # current version is 1
+    "ecu_id": "autoware",  # should be unique for each ECU in vehicle
+}
 
-    def get_ecu_id(self):
-        return self._ecu_info["ecu_id"]
 
-    def get_ecu_ip_addr(self):
-        return self._ecu_info.get("ip_addr", "localhost")
+@dataclass
+class ECUInfo:
+    """
+    Version 1 scheme example:
+        format_vesrion: 1
+        ecu_id: "autoware"
+        ip_addr: "0.0.0.0"
+        bootloader: "grub"
+        secondaries:
+            - ecu_id: "p1"
+              ip_addr: "0.0.0.0"
+        available_ecu_ids:
+            - "autoware"
+            - "p1
+    """
 
-    def get_available_ecu_ids(self):
-        return self._ecu_info.get("available_ecu_ids", [self.get_ecu_id()])
+    ecu_id: str
+    ip_addr: str = "127.0.0.1"
+    bootloader: str = BootloaderType.UNSPECIFIED.value
+    available_ecu_ids: list = field(default_factory=list)  # list[str]
+    secondaries: list = field(default_factory=list)  # list[dict[str, Any]]
+    format_version: int = 1
 
-    def _load_ecu_info(self, path: str):
+    @classmethod
+    def parse_ecu_info(cls, ecu_info_file: Union[str, Path]) -> "ECUInfo":
+        ecu_info = DEFAULT_ECU_INFO.copy()
         try:
-            with open(path) as f:
-                ecu_info = yaml.safe_load(f)
+            ecu_info = yaml.safe_load(Path(ecu_info_file).read_text())
+            assert isinstance(ecu_info, Dict)
         except Exception:
-            return EcuInfo.DEFAULT_ECU_INFO
-        return ecu_info
+            logger.warning(
+                f"failed to load {ecu_info_file=} or config file corrupted, use default config"
+            )
+        logger.info(f"ecu_info={ecu_info}")
+
+        # load options
+        # NOTE: if option is not presented,
+        #       this option will be set to the default value
+        _ecu_info_dict: Dict[str, Any] = dict()
+        for _field in fields(cls):
+            _option = ecu_info.get(_field.name)
+            if not isinstance(_option, _field.type):
+                if _option is not None:
+                    logger.warning(
+                        f"{_field.name} contains invalid value={_option}, "
+                        "ignored and set to default={_field.default}"
+                    )
+                if _field.default is MISSING and _field.default_factory is MISSING:
+                    raise ValueError(
+                        f"required field {_field.name} is not presented, abort"
+                    )
+                _ecu_info_dict[_field.name] = (
+                    _field.default
+                    if _field.default is not MISSING
+                    else _field.default_factory()  # type: ignore
+                )
+                continue
+            # parsed _option is available
+            if isinstance(_option, (list, dict)):
```

Comment:
if-elif-else is prefer
```suggestion
            # parsed _option is available
            elif isinstance(_option, (list, dict)):
```